File: solutions/c/grains/1/grains.c
#include "grains.h"

uint64_t square(uint8_t index){
    if (index == 0) return 0;
    uint64_t g = 1;
    for (int i = 2; i <= index; i++) g *= 2;
    return g;
}
uint64_t total(void){
    uint64_t sum = 0;
    for (int i = 1; i <= 64; i++) sum += square(i);
    return sum;
}

File: solutions/python/grains/1/grains.py
def square(number):
    if number < 1 or number > 64:
        raise ValueError("square must be between 1 and 64")
    grains = 1
    number -= 1
    while number > 0:
        grains *= 2
        number -= 1
    return grains

def total():
    grains = 0
    for n in range(1, 65):
        grains += square(n)
    return grains
